src/OpenLoco/GameCommands/GameCommands.h

```diff
@@ -445,31 +445,71 @@ namespace OpenLoco::GameCommands
         return doCommand(GameCommand::removeTown, regs) != FAILURE;
     }
 
-    // Remove company headquarters (or build - needs to be checked)
-    // Note: The game seems to call do_55, then do_54 in success case and returns 0x0 code
-    //       if we try to build over the existing only do_54 is called and it returns 0x80000000 (in regs.ebx)
-    inline uint32_t do_54(uint8_t bl, uint16_t ax, uint16_t cx, uint16_t di, uint16_t dx)
+    struct HeadquarterPlacementArgs
     {
-        registers regs;
+        HeadquarterPlacementArgs() = default;
+        explicit HeadquarterPlacementArgs(const registers regs)
+            : pos(regs.ax, regs.cx, regs.di)
+            , rotation(regs.bh & 0x3)
+            , type(regs.dl)
+            , buildImmediately(regs.bh & 0x80)
+        {
+        }
+
+        Map::Pos3 pos;
+        uint8_t rotation;
+        uint8_t type;
+        bool buildImmediately = false; // No scaffolding required (editor mode)
+        explicit operator registers() const
```

Comment:
I've no real problem with this, but I'm wondering whether it might be better to pass an existing `registers` struct instead. Presumably, it's a conscious design decision, so could you elaborate on the choice?

Reply:
How would that work? It wouldn't be a conversion operator then. Yes you could make it a normal method exportToRegisters but just felt like a conversion operator would be a fun way of doing it